code/modules/antagonists/heretic/magic/void_conduit.dm

```diff
@@ -0,0 +1,115 @@
+/datum/action/cooldown/spell/conjure/void_conduit
+	name = "Void Conduit"
+	desc = "Opens a gate to the Void; it releases an intermittent pulse that damages windows and airlocks, \
+		while afflicting Heathens with  void chill. \
+		Heretics receive are granted the cold resistance and low pressure resistance."
+	background_icon_state = "bg_heretic"
+	overlay_icon_state = "bg_heretic_border"
+	button_icon = 'icons/mob/actions/actions_ecult.dmi'
+	button_icon_state = "void_rift"
+
+	cooldown_time = 1 MINUTES
+
+	sound = null
+	school = SCHOOL_FORBIDDEN
+	invocation = "MBR'C' TH' V''D!"
+	invocation_type = INVOCATION_SHOUT
+	spell_requirements = NONE
+
+	summon_radius = 0
+	summon_type = list(/obj/structure/void_conduit)
+	summon_respects_density = TRUE
+	summon_respects_prev_spawn_points = TRUE
+
+/obj/structure/void_conduit
+	name = "Void Conduit"
+	desc = "An open gate which leads to nothingness. Releases pulses which you do not want to get hit by."
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "void_conduit"
+	anchored = TRUE
+	density = TRUE
+	///Overlay to apply to the tiles in range of the conduit
+	var/static/image/void_overlay = image(icon = 'icons/turf/overlays.dmi', icon_state = "voidtile")
+	///How many tiles far our effect is
+	var/effect_range = 12
+	///id of the deletion timer
+	var/timerid
+	///Audio loop for the rift being alive
+	var/datum/looping_sound/void_conduit/soundloop
+
+/obj/structure/void_conduit/Initialize(mapload)
+	. = ..()
+	soundloop = new(src, start_immediately = TRUE)
+	timerid = QDEL_IN_STOPPABLE(src, 1 MINUTES)
+	START_PROCESSING(SSobj, src)
+	for(var/turf/affected_turf in RANGE_TURFS(effect_range, src))
+		if(!isopenturf(affected_turf))
+			continue
+		affected_turf.add_overlay(void_overlay)
+		void_overlay.mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+		void_overlay.alpha = 180
+
+/obj/structure/void_conduit/Destroy(force)
+	QDEL_NULL(soundloop)
+	deltimer(timerid)
+	STOP_PROCESSING(SSobj, src)
+	for(var/turf/affected_turf in RANGE_TURFS(effect_range, src))
+		affected_turf.cut_overlay(void_overlay)
```

Comment:
What happens if the conduit is moved, (i know it's probably not movable under normal circumstances but you never know) do the overlays get stuck, since you're not keeping a running list of all affected turfs?

Reply:
Also
```suggestion
	for(var/turf/affected_turf as anything in RANGE_TURFS(effect_range, src))
		affected_turf.cut_overlay(void_overlay)
```